### PR TITLE
feat: add outputs for github repo scaffolding plugin

### PR DESCRIPTION
**Pull request by IronCore864**
# Summary

- docs update for argocdapp/github scaffolding/github actions plugin
- add outputs for github actions plugin
- backwards compatible/non-breaking change (tested, state/core not affected by added output)

## Key Points

- [x] Documentation (new or existing) is updated.

### CODE REVIEW

File: docs/plugins/githubactions-golang_plugin.md [hunk 1]
```diff
@@ -65,3 +65,55 @@ tools:
...
+    branch: main
+    image_repo: ironcore864/go-webapp-devstream-demo
+- name: golang-demo-actions
+  plugin:
+    kind: githubactions-golang
+    version: 0.2.0
+  dependsOn: ["go-webapp-repo.github-repo-scaffolding-golang"]
+  options:
+    owner: ${{go-webapp-repo.github-repo-scaffolding-golang.outputs.owner}}
+    repo: ${{go-webapp-repo.github-repo-scaffolding-golang.outputs.repo}}
```
Comment (lfbdev) [inline docs/plugins/githubactions-golang_plugin.md:99]: Should be filled in by the consumer plugin,  this is the producer plugin

Comment (lfbdev) [inline docs/plugins/githubactions-golang_plugin.md:99]: Should be filled in by the consumer plugin,  this is the producer plugin

File: docs/plugins/github-repo-scaffolding-golang_plugin.md [hunk 2]
```diff
@@ -43,4 +43,59 @@ tools:
...
+    branch: main
+    image_repo: ironcore864/go-webapp-devstream-demo
+- name: golang-demo-actions
+  plugin:
+    kind: githubactions-golang
+    version: 0.2.0
+  dependsOn: ["go-webapp-repo.github-repo-scaffolding-golang"]
+  options:
+    owner: ${{go-webapp-repo.github-repo-scaffolding-golang.outputs.owner}}
+    repo: ${{go-webapp-repo.github-repo-scaffolding-golang.outputs.repo}}
```
Comment (lfbdev) [inline docs/plugins/github-repo-scaffolding-golang_plugin.md:86]: Should be filled in by the consumer plugin,  this is the producer plugin.

Comment (lfbdev) [inline docs/plugins/github-repo-scaffolding-golang_plugin.md:86]: Should be filled in by the consumer plugin,  this is the producer plugin.

File: docs/plugins/argocdapp_plugin.md [hunk 3]
```diff
@@ -43,3 +43,51 @@ tools:
...
+    app:
+      name: hello
+      namespace: argocd
+    destination:
+      server: https://kubernetes.default.svc
+      namespace: default
+    source:
+      valuefile: values.yaml
+      path: charts/go-hello-http
+      repoURL: ${{go-webapp-repo.github-repo-scaffolding-golang.outputs.repoURL}}
```
Comment (lfbdev) [inline docs/plugins/argocdapp_plugin.md:85]: Should be filled in by the consumer plugin, this is the producer plugin.

File: docs/plugins/argocdapp_plugin.md [hunk 4]
```diff
@@ -43,3 +43,51 @@ tools:
...
+```
+
+In the example above:
+
+- We put `go-webapp-repo.github-repo-scaffolding-golang` as dependency by using the `dependsOn` keyword.
+- We used `go-webapp-repo.github-repo-scaffolding-golang`'s output as input for the `githubactions-golang` plugin.
```
Comment (lfbdev) [inline docs/plugins/argocdapp_plugin.md:91]: used -- use

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**